action.yml

```
@@ -72,45 +72,65 @@ runs:
       shell: bash
       run: |
         aws ecs update-service \
-          --health-check-grace-period-seconds 900 \
           --task-definition "${{ steps.deploy-task-definition.outputs.latest-task-definition-arn }}" \
+          --desired-count 1 \
           --cluster ${{ inputs.environment }} \
           --service ${{ inputs.ecr_repository }}-${{ inputs.environment }} \
-          --desired-count 1 \
           --region ${{ inputs.aws_region }}
 
     - name: Await service stability
       shell: bash
-      # ToDo: MD-7199 Re-evaluate logic after deployment failures are correctly handled by ECS
       id: check-service-stability
       run: |
         aws ecs wait services-stable \
           --cluster ${{ inputs.environment }} \
           --service ${{ inputs.ecr_repository }}-${{ inputs.environment }} \
           --region ${{ inputs.aws_region }}
 
-    - name: Deregister previous local-exec task definition
-      if: ${{ always() && steps.deploy-local-task-definition.outputs.previous-task-definition-arn != '' }}
+    - name: Determine which task definition arn should be deregistered
+      if: always()
       shell: bash
+      id: determine-arn-to-deregister
       run: |
-        if [ "${{ steps.check-service-stability.outcome }}" == 'success' ]; then
-            TASK_DEFINITION_TO_DEREGISTER="${{ steps.deploy-local-task-definition.outputs.previous-task-definition-arn }}"
+        PRIMARY_DEPLOYMENT_DEFINITION_ARN=$(
+          aws ecs describe-services \
+            --cluster ${{ inputs.environment }} \
+            --services ${{ inputs.ecr_repository }}-${{ inputs.environment }} \
+            --region ${{ inputs.aws_region }} | jq -r '.services[].deployments[] | select(.status == "PRIMARY") | .taskDefinition'
+        )
+        PREVIOUS_TASK_DEFINITION="${{ steps.deploy-task-definition.outputs.previous-task-definition-arn }}"
+        LATEST_TASK_DEFINITION="${{ steps.deploy-task-definition.outputs.latest-task-definition-arn }}"
+        
+        if [[ "$PRIMARY_DEPLOYMENT_DEFINITION_ARN" == "$LATEST_TASK_DEFINITION" ]]; then
+          if [ ! -z "$PREVIOUS_TASK_DEFINITION" ]; then
+             echo "arn-to-deregister=previous" >> $GITHUB_OUTPUT
+          fi
         else
-            TASK_DEFINITION_TO_DEREGISTER="${{ steps.deploy-local-task-definition.outputs.latest-task-definition-arn }}"
+          echo "arn-to-deregister=latest" >> $GITHUB_OUTPUT
```

```suggestion
          echo "revision-to-deregister=latest" >> $GITHUB_OUTPUT
```
Here, "revision" would make a lot of sense to me.